### PR TITLE
Update: Fleet.Vehicles to 2.0.0

### LINKED ISSUE #512: [Model Update]: Update io.catenax.fleet.vehicles to version 2.0.0
## Update Reason
Use quality decided to put many of the current properties to optional for CX 24.05 release. In addition, some typos in the model example payload and in the description exists.
Currently there is a link o another semantic model io.catenax.vehicle.product_description:3.0.0 -> use case quality has not yet finally decided to integrate this model and to get rid of  the dependency or keep the existing structure.

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists - Catena-X quality use case
- [x] The potential updates are discussed with all stakeholders - discussed in Catena-X use case quality
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [X] The potential update will be either 
  - [X] backward compatible or 
  - [X] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by TorstenHirsch**
## Description

QAX (Catena-X use case quality) decided to set most of the properties to optional and also to streamline the aspect data model.
### Added
- metaInformation property added to root entity
- integration of the shared UUID characteristic and RegEx for the vehicle Catena-X identifie
- Introducing new characteristic for unique identifiers
- integration of separate aspect model Vehicle.ProductDescription:3.0.0 into this model to do necessary adaptions that were decided in Catena-X Quality use case team: Most of the properties should be set to optional
- meta data entity: Gives data consumer necessary information how data provider created this data set

### Changed
- changed all properties to optional except one unique identifier per entity
- properties that were linked to OEM entity are now one level up 
- fuel type property now linked to engine property
- entity production was renamed to productionPlant to increase clarity

### Removed
- link to Vehicle.ProductDescription:3.0.0: Previous linked model now integrated and adapted
Streamlining of aspect model:
- remove of body entity + body properties as it was never used
- remove of all KBA properties: Focusing on NHTSA properties only 
- remove of oem entity: Properties now located on vehicle level
- remove of fuel entity: Properties now located on vehicle and engines level
- remove of sale entity: Properties now located on vehicle level
- remove of production entity: Properties now located on vehicle level

Closes #512 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by TorstenHirsch** [comment]
@agg3fe modelling_team

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+   samm:dataType xsd:string .
+
+:CountryCodeRegularExpression a samm-c:RegularExpressionConstraint ;
+   samm:preferredName "Country Code Regular Expression"@en ;
+   samm:description "Regular Expression that ensures a three-letter code "@en ;
+   samm:see <https://www.iso.org/iso-3166-country-codes.html> ;
+   samm:value "^[A-Z][A-Z][A-Z]$" .
+
+:WorldManufacturerInformationCodeLength a samm-c:LengthConstraint ;
+   samm:description "Restricts the length of wmiCode to exactly 3 chars."@en ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:296]: `preferredName` is missing

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+:RecordStatusEnumeration a samm-c:Enumeration ;
+   samm:preferredName "Record status enumeration"@en ;
+   samm:description "Defining the enumeration for the record status"@en ;
+   samm:dataType xsd:string ;
+   samm-c:values ( "new" "update" "delete" "same" ) .
+
+:UniqueID a samm:Characteristic ;
+   samm:preferredName "unique ID"@en ;
+   samm:description "This Characteristic describes a property that is truly unque either inside the Company that is providing the data or even in Catena-X dataspace. Only one property in a semantic entity should be a uniqueID. Unique ids can be used to link a data set with other semantic models."@en ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:237]: typo `unque`

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+   samm:description "This entity groups all meta information properties. Meta information properties give hints for a Catena-X data consumer to better understand, how this data set was created on the Catena-X data provider side."@en ;
+   samm:properties ( :selectionCriteria [ samm:property :selectionStart; samm:optional true ] [ samm:property :selectionEnd; samm:optional true ] ) .
+
+:recordStatus a samm:Property ;
+   samm:preferredName "Record status"@en ;
+   samm:description "The record operation enunumeration can be used to realize delta update concept.\nDelta updateupdate concept means:\n- You transfer an initial load of data\n- After the first week only the delta to the initial load is transferred\n\n\nThe record status describes whether this record is:\n- new=>This record is transferred the first time\n- update=> Some properties of this record have changed compared to a previous transfer\n- delete=> This record was transferred in the initial load or in a previous delta update, but is not \nused any more and therefore it should be deleted on data consumer side\n- same=> This record was transferred in the initial load or in a previous delta update"@en ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:70]: typo `enunumeration`, `updateupdate` , `\nused`.
please check in other models also if used same description.

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+   samm:characteristic :MetaCharacteristic .
+
+:ListOfVehicles a samm-c:List ;
+   samm:preferredName "List of vehicles"@en ;
+   samm:description "List of vehicles"@en ;
+   samm:dataType :Vehicle .
+
+:MetaCharacteristic a samm:Characteristic ;
+   samm:preferredName "MetaCharacteristic"@en ;
+   samm:description "Chracteristic for the meta information."@en ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:55]: typo `Chracteristic`

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#> .
+
+:Vehicles a samm:Aspect ;
+   samm:preferredName "Vehicles"@en ;
+   samm:description "This aspect model was created in the Catena-X use case quality. The purpose of this aspect model is to transfer general information of a list/fleet of vehicles - primarily how the vehicles are equipped, when they were built and which engines are installed."@en ;
+   samm:properties ( :listOfVehicles [ samm:property :metaInformation; samm:optional true ] ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:listOfVehicles a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:38]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+   samm:exampleValue "WVW" .
+
+:wmiNameNHTSA a samm:Property ;
+   samm:preferredName "WMI name according NHTSA"@en ;
+   samm:description "Name of the vehicle manufacturer according NHTSA: use https://vpic.nhtsa.dot.gov/api/vehicles/decodewmi/WVW?format=json - entry to be used is CommonName."@en ;
+   samm:see <https://vpic.nhtsa.dot.gov/> ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "Volkswagen" .
+
+:engines a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:201]: Property and the referenced Characteristic should not have the same name

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+   samm:preferredName "Engines"@en ;
+   samm:description "List of installed engines in the vehicle."@en ;
+   samm:characteristic :Engines .
+
+:equipments a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:206]: Property and the referenced Characteristic should not have the same name

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl (hunk 5 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:38]: Renamed to VehiclesList

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl (hunk 6 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:201]: Renamed to EnginesList

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl (hunk 7 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:206]: Changed to EquipmentsList

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl (hunk 2 above)
Comment (TorstenHirsch) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:237]: fixed

**Comment by TorstenHirsch** [comment]
@agg3fe Please check - I fixed all mentioned findings
modelling_team

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+   samm:description "This entity groups all meta information properties. Meta information properties give hints for a Catena-X data consumer to better understand, how this data set was created on the Catena-X data provider side."@en ;
+   samm:properties ( :selectionCriteria [ samm:property :selectionStart; samm:optional true ] [ samm:property :selectionEnd; samm:optional true ] ) .
+
+:recordStatus a samm:Property ;
+   samm:preferredName "Record status"@en ;
+   samm:description "The record operation enunumeration can be used to realize delta update concept.\nDelta update concept means:\n- You transfer an initial load of data\n- After the first week only the delta to the initial load is transferred\n\n\nThe record status describes whether this record is:\n- new=>This record is transferred the first time\n- update=> Some properties of this record have changed compared to a previous transfer\n- delete=> This record was transferred in the initial load or in a previous delta update, but is not \nused any more and therefore it should be deleted on data consumer side\n- same=> This record was transferred in the initial load or in a previous delta update"@en ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:70]: please fix typo `enunumeration`. should be enumeration.

**Comment by agg3fe** [comment]
@TorstenHirsch please fix one typo, then we are good for MS2.

File: io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl (hunk 8 above)
Comment (agg3fe) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl:70]: @TorstenHirsch  please fix this typo in this and other model.

Comment (TorstenHirsch) [reply io.catenax.fleet.vehicles/2.0.0/Vehicles.ttl]: Fixed @agg3fe  
modelling_team

**Pull request merged**